sw/host/hsmtool/src/spxef/mod.rs

```diff
@@ -0,0 +1,243 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use acorn::{GenerateFlags, KeyEntry, KeyInfo, SpxInterface};
+use anyhow::Result;
+use cryptoki::session::Session;
+use sphincsplus::{DecodeKey, EncodeKey};
+use sphincsplus::{SphincsPlus, SpxDomain, SpxError, SpxPublicKey, SpxSecretKey};
+use std::rc::Rc;
+use std::str::FromStr;
+
+use crate::error::HsmError;
+use crate::util::attribute::{AttrData, AttributeMap, AttributeType};
+use crate::util::ef::ElementaryFile;
+
+/// Shredder holds a string and destroys it upon drop.
+pub struct Shredder(String);
+
+impl Shredder {
+    pub fn as_str(&self) -> &str {
+        self.0.as_str()
+    }
+    pub fn as_bytes(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+}
+
+impl Drop for Shredder {
+    fn drop(&mut self) {
+        unsafe {
+            for v in self.0.as_mut_vec().iter_mut() {
```

Comment:
Is this unsafe because you want to be able to use `Shredder` as a constant variable?

Reply:
Its unsafe because `String::as_mut_vec` allows you to modify individual bytes of a UTF-8 string, thus leading to the possibility of creating an illegal UTF-8 code-point.  It's moot because we're immediately dropping the string afterwards.

I've refactored Shredder into a more generic container type that can hold any type that implements `Shreddable`.  This restructuring will come in handy in fixing other bits of the code that need to destroy upon drop.